src/main/scala/com/qubole/spark/hiveacid/writer/hive/HiveAcidWriter.scala

```diff
@@ -221,12 +221,33 @@ private[writer] class HiveAcidFullAcidWriter(options: WriterOptions,
       }
     }
 
+    def safeDeletePath(fs: FileSystem, path: Path): Boolean = {
+      try {
+        fs.delete(path, false)
```

Comment:
@amoghmargoor : Should we do this ? 

`if(fs.exists(path)) {
    fs.delete(path, false)
 }`

Since fs.exist() does filestatus() call whereas delete() first calls filestatus() and then delete(). Assuming that in most of the cases task retries won't happen - we would only end up making getFileStatus().

Reply:
done.